### PR TITLE
Feature/comments

### PR DESCRIPTION
**Pull request by tubhutani13**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -7,6 +7,7 @@ class User < ApplicationRecord
 
   has_many :questions, dependent: :nullify
   has_many :answers, dependent: :nullify
+  has_many :comments, dependent: :nullify
```
Comment (shubhamjuneja1) [inline app/models/user.rb:15]: should be restrict_with_error

File: app/models/answer.rb [hunk 2]
```diff
@@ -2,6 +2,8 @@ class Answer < ApplicationRecord
   belongs_to :user
   belongs_to :question
 
+  has_many :comments, as: :commentable
```
Comment (shubhamjuneja1) [inline app/models/answer.rb]: use concern and include in every entity having comments for comment related associations and methods

Comment (shubhamjuneja1) [reply app/models/answer.rb]: dependent:  ?

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  def comment_params
+    params.require(:comment).permit(:user_id, :body, :commentable_id)
+  end
+
+  def set_comment
+    @comment = Comment.find(params[:id])
+  end
+
+  def set_entity
```
Comment (shubhamjuneja1) [inline app/controllers/comments_controller.rb:40]: In params, pass commentable_name and commentable_id and load that

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
+class CommentsController < ApplicationController
+  before_action :set_comment, only: [:destroy]
+  before_action :set_entity
+
+  def create
+    @comment = @entity.comments.new(comment_params)
+    @comment.user = current_user
```
Comment (shubhamjuneja1) [inline app/controllers/comments_controller.rb]: merge in params

File: app/controllers/comments_controller.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    if @comment.save
+      redirect_to @question, notice: "Comment submitted successfully"
+    else
+      flash[:error] = "error while creating comment"
+      render "questions/show", status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @comment.destroy
```
Comment (shubhamjuneja1) [inline app/controllers/comments_controller.rb]: check permssion
if fails?

File: app/controllers/comments_controller.rb [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    redirect_to @question, notice: "Comment Deleted successfully"
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:user_id, :body, :commentable_id)
+  end
+
+  def set_comment
```
Comment (shubhamjuneja1) [inline app/controllers/comments_controller.rb]: rename load_comment

File: app/helpers/comments_helper.rb [hunk 7]
```diff
@@ -0,0 +1,10 @@
+module CommentsHelper
+    def comment_custom_path(comment,entity,question)
```
Comment (shubhamjuneja1) [inline app/helpers/comments_helper.rb:2]: why need this?
aere we using a separate page?

**Review by shubhamjuneja1 — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 1 above)
Comment (tubhutani13) [reply app/models/user.rb:15]: please discuss

File: app/helpers/comments_helper.rb (hunk 7 above)
Comment (tubhutani13) [reply app/helpers/comments_helper.rb:2]: to handle requests

**Pull request merged**